Create a VM on an Azure Arc-enabled vSphere cluster (#1233)
microsoft/azure_arc: docs/azure_arc_jumpstart/azure_arc_vsphere/day2/vmware_arm_template_win/_index.md

  ```shell
  az --version
  ```

- Create Azure service principal (SP).

    To be able to complete the scenario and its related automation, Azure service principal assigned with the “Contributor” role is required. To create it, login to your Azure account run the below command (this can also be done in [Azure Cloud Shell](https://shell.azure.com/)).

    ```shell
    az login
    subscriptionId=$(az account show --query id --output tsv)
    az ad sp create-for-rbac -n "<Unique SP Name>" --role "Contributor" --scopes /subscriptions/$subscriptionId
    ```

    For example:

    ```shell
    az login
    subscriptionId=$(az account show --query id --output tsv)
    az ad sp create-for-rbac -n "JumpstartArc" --role "Contributor" --scopes /subscriptions/$subscriptionId
    ```

    Output should look like this:

    ```json
    {
    "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "displayName": "JumpstartArc",
    "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    }
    ```

>     > **NOTE: It is optional but highly recommended to scope the SP to a specific [Azure subscription and resource group](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest)**

Reviewer: ```suggestion
    > **NOTE: It is optional but highly recommended to scope the SP to a specific [Azure subscription and resource group](https://docs.microsoft.com/cli/azure/ad/sp?view=azure-cli-latest)**
```

2. User deploys the ARM template at the resource group level.

3. User verifies the deployment of the Windows virtual machine in the vSphere cluster.

## Create the Virtual Machine

As mentioned, this deployment will leverage an ARM template. You will deploy a single ARM template at resource group scope.

- Before deploying the ARM template, login to Azure using AZ CLI with the ```az login``` command.

- The deployment will use an ARM template parameters file to customize your environment. Before initiating the deployment, edit the [_azuredeploy.parameters.json_](https://github.com/microsoft/azure_arc/blob/main/azure_arc_vsphere_jumpstart/vmware_arm_template_win/azuredeploy.parameters.json) file located in your local cloned repository folder. Example parameters files are located [here](https://github.com/microsoft/azure_arc/blob/main/azure_arc_vsphere_jumpstart/vmware_arm_template_win/azuredeploy.example.parameters.json). Fill out the parameters according to your environment:

  - _`virtualMachineName`_ - the name of the Windows virtual machine that will be created

>   - _`extendedLocationName`_ - the name of your custom location

Reviewer: ```suggestion
  - _`extendedLocationName`_ - the name of your Azure Arc custom location
```

  - _`vCenterName`_ - the name of your vCenter

>   - _`networkName`_ - the network you will use to deploy your VM

Reviewer: ```suggestion
  - _`networkName`_ - the vSphere network (port group) you will use to deploy your VM
```

  - _`nicType`_ - the network type to associate to the VM
  - _`templateName`_ - your Windows VM template name
  - _`clusterName`_ - the vSphere cluster to host the VM

>   - _`adminUserName`_ - the admin user name for the created Windows VM

Reviewer: ```suggestion
  - _`adminUserName`_ - the admin username for the created Windows VM
```

  - _`adminUserPassword`_ - the admin user password for the created Windows VM

>   

Reviewer: Please update where applicable in the first VMware scenario
